File: ECE39500/exams/bak_midterm-2/PartA/EofState.java
public class EofState extends CompilerState {

   Compiler compiler = null; 

   public EofState(Compiler compiler) {
      this.compiler = compiler;
   }

   public void comma( ) { 
      System.out.println("Parsing EofState, expected eol, found ,");
      compiler.state = StateFactory.getState("ErrorState");
   }

   public void value(String str) {
      System.out.println("Parsing EofState, expected eol, found value");
      compiler.state = StateFactory.getState("ErrorState");
   }

   public void eol( ) { }
      
   public void eof( ) {
      System.out.println("Parsing EofState, expected value, found eof");
      compiler.state = StateFactory.getState("ErrorState");
   }
   public String toString( ) {
      return "EofState";
   }
}

File: ECE39500/exams/bak_midterm-2/midterm-2/PartA/EofState.py
from CompilerState import CompilerState, StateFactory
import sys

class EofState(CompilerState):
    INT_MIN = -sys.maxsize - 1

    def __init__(self, compiler):
        self.compiler = compiler
        return
    
    def comma(self):
        print("Parsing EofState, expected eol, found ,")
        self.compiler.state = StateFactory.getState("ErrorState")

    def value(self):
        print("Parsing EofState, expected eol, found value")
        self.compiler.state = StateFactory.getState("ErrorState")

    def eol(self):
        pass

    def eof(self):
        print("Parsing EofState, expected value, found eof")
        self.compiler.state = StateFactory.getState("ErrorState")
    
    def __str__(self):
        return "EofState"